src/main/java/com/epam/reportportal/extension/azure/AzureExtension.java

```diff
@@ -240,39 +283,87 @@ public boolean testConnection(Integration integration) {
 
 	@Override
 	public Optional<Ticket> getTicket(String id, Integration integration) {
-		IntegrationParameters params = getParams(integration);
-		ApiClient defaultClient = getConfiguredApiClient(params.getPersonalAccessToken());
-		String organizationName = extractOrganizationNameFromUrl(defaultClient, params.getOrganizationUrl());
+		initFields(integration);
 
 		WorkItemsApi workItemsApi = new WorkItemsApi(defaultClient);
 		try {
 			WorkItem workItem = workItemsApi
 					.workItemsGetWorkItem(organizationName, Integer.valueOf(id), params.getProjectName(), API_VERSION,
 							null, null, null);
-			Ticket ticket = new Ticket();
-			ticket.setId(workItem.getId().toString());
-			ticket.setTicketUrl(workItem.getUrl());
-			ticket.setStatus(workItem.getFields().get("System.State").toString());
-			ticket.setSummary(workItem.getFields().get("System.Title").toString());
-			return Optional.of(ticket);
+			return Optional.of(convertWorkItemToTicket(workItem));
 		} catch (ApiException e) {
 			LOGGER.error("Unable to load ticket: " + e.getMessage(), e);
 			return Optional.empty();
 		}
 	}
 
 	@Override
-	// TODO: Implement in the future
+	// TODO: Add attachments from the test-item to the issue
 	public Ticket submitTicket(PostTicketRQ ticketRQ, Integration integration) {
-		return null;
+		initFields(integration);
+
+		List<JsonPatchOperation> patchOperationList = new ArrayList<>();
+
+		uploadAttachmentToAzure(ticketRQ);
+
+		String issueType = null;
+		List<PostFormField> fields = ticketRQ.getFields();
+		for (PostFormField field : fields) {
+			String id = replaceSeparators(field.getId());
+			String operation = "add";
+			String path = "/fields/" + id;
+			String value = field.getValue().get(0);
+
+			if ("issuetype".equals(field.getId())) {
+				issueType = field.getValue().get(0);
+				continue;
+			}
+			if ("System.Description".equals(id)) {
+				path = "/fields/System.Description";
+				value = field.getValue().get(0) + getDescriptionFromTestItem(ticketRQ);
+			}
+			patchOperationList.add(new JsonPatchOperation(null, operation, path, value));
+		}
+
+		WorkItemsApi workItemsApi = new WorkItemsApi(defaultClient);
+		WorkItem workItem = null;
+		List<JsonPatchOperation> patchOperationsForAttachment = new ArrayList<>();
+
+		try {
+			workItem = workItemsApi
+					.workItemsCreate(organizationName, patchOperationList, params.getProjectName(), issueType,
+							API_VERSION, null, null, null, null);
+
+			if (!attachmentsURL.isEmpty()) {
+				String operation = "add";
```

Comment:
function is a bit long. please make smth like getPatchOperaionsForAttachments and then update if it's not null and empty

Reply:
Did it.